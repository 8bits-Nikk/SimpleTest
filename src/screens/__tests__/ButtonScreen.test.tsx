import {renderWithNavigation} from '../../helper/renderWithNavigation';
import ButtonScreen from '../ButtonScreen';
import {act, fireEvent, waitFor} from '@testing-library/react-native';

describe('Button Screen Tests', () => {
  it('should render correctly', () => {
    const tree = renderWithNavigation(<ButtonScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should have button', () => {
    const tree = renderWithNavigation(<ButtonScreen />);
    const Btn_1 = tree.queryByTestId('Button_1');
    const Btn_2 = tree.queryByTestId('Button_2');
    const Btn_3 = tree.queryByTestId('Button_3');
    expect(Btn_1).not.toBeNull();
    expect(Btn_2).not.toBeNull();
    expect(Btn_3).not.toBeNull();
  });

  it('should fire onPress', async () => {
    const tree = renderWithNavigation(<ButtonScreen />);
    const Btn_1 = tree.getByTestId('Button_1');
    const Btn_2 = tree.getByTestId('Button_2');
    const Btn_3 = tree.getByTestId('Button_3');

    const text = tree.queryByText(' Pressed');

    await act(async () => {
      await fireEvent.press(Btn_1);
      expect(text?.props.children).toBe('Button 1 Pressed');
      await fireEvent.press(Btn_2);
      expect(text?.props.children).toBe('Button 2 Pressed');
      await fireEvent.press(Btn_3);
      expect(text?.props.children).toBe('Button 3 Pressed');
    });
  });

  // Second method for async state update
  it('should fire onPress2', async () => {
    const tree = renderWithNavigation(<ButtonScreen />);
    const Btn_1 = tree.getByTestId('Button_1');
    const Btn_2 = tree.getByTestId('Button_2');
    const Btn_3 = tree.getByTestId('Button_3');

    const text = tree.queryByText(' Pressed');

    await waitFor(() => {
      fireEvent.press(Btn_1);
      expect(text?.props.children).toBe('Button 1 Pressed');
      fireEvent.press(Btn_2);
      expect(text?.props.children).toBe('Button 2 Pressed');
      fireEvent.press(Btn_3);
      expect(text?.props.children).toBe('Button 3 Pressed');
    });
  });
});
