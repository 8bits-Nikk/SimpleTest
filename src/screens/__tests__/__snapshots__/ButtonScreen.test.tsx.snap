// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Screen Tests should render correctly 1`] = `
<RCTSafeAreaView
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text>
       Pressed
    </Text>
    <View
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="Button_1"
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Button 1
        </Text>
      </View>
    </View>
    <View
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="Button_2"
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Button 2
        </Text>
      </View>
    </View>
    <View
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="Button_3"
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Button 3
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#ddd",
        "flexDirection": "row",
        "justifyContent": "space-evenly",
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="Previous_btn"
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          disabled={false}
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Previous
        </Text>
      </View>
    </View>
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="Next_btn"
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          disabled={false}
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Next
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
